### PR TITLE
add new package qlik-visualizationextensions

### PR DESCRIPTION
**Pull request by konne**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `npm run new-package package-name`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

**Comment by konne** [comment]
@minestarks do I have to do more, that this request get merged?

File: qlik-visualizationextensions/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,2417 @@
...
+
+        /**
+         * A value.
+         * This parameter is optional.
+         */
+        qNum?: number;
+
+        /**
+		 * Rank number of the value, starting from 0.
+         * If the element number is a negative number, it means that the returned value is not an element number.
```
Comment (minestarks) [inline qlik-visualizationextensions/index.d.ts]: odd indentation here

File: qlik-visualizationextensions/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6",
+            "dom"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": false,
+        "baseUrl": "../",
```
Comment (minestarks) [inline qlik-visualizationextensions/tsconfig.json]: New libraries should have ```strictNullChecks``` on.

File: qlik-visualizationextensions/tslint.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{ "extends": "../tslint.json",
+  "rules": {
```
Comment (minestarks) [inline qlik-visualizationextensions/tslint.json]: Add newline after ```{```

**Review by minestarks — CHANGES_REQUESTED** [review]

File: qlik-visualizationextensions/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,2417 @@
...
+         */
+        qSelectionThreshold: number;
+
+        /**
+         * Object with number of values in different states.
+         */
+        qStateCounts: any;
+
+        /**
+         * Concatenated string of selected values if number of values are less than the threshold, or if the string is of fo
```
Comment (konne) [inline qlik-visualizationextensions/index.d.ts]: @minestarks sorry I don't understand that comment to that line. Could it be that you mean another line in the code, please give me a hint

File: qlik-visualizationextensions/index.d.ts (hunk 1 above)
Comment (konne) [inline qlik-visualizationextensions/index.d.ts]: @minestarks 
what is the prefered maximum col count?

**Review by konne — COMMENTED** [review]
@minestarks please check my changes if they fix your mentioned issues

**Comment by mhegazy** [comment]
Some lint failures need to be addressed:

> Error in qlik-visualizationextensions
Command failed: node /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/node_modules/tslint/lib/index.js/../tslint-cli.js --format stylish index.d.ts qlik-visualizationextensions-tests.ts
index.d.ts
1301:1  max-line-length  Exceeds maximum line length of 200
1302:1  max-line-length  Exceeds maximum line length of 200
1323:1  max-line-length  Exceeds maximum line length of 200
1324:1  max-line-length  Exceeds maximum line length of 200

**Review by minestarks — APPROVED** [review]

**Comment by konne** [comment]
@minestarks this is my first PR to DT, do I have todo anything after you "approved" this PR, or is this automatically merged to the master branch

**Pull request merged**